windows/shared/logging/PlogTraceLogging/include/PlogTraceLogging/PlogTraceLogging.hxx

```diff
@@ -0,0 +1,46 @@
+
+#ifndef PLOG_TRACE_LOGGING_HXX
+#define PLOG_TRACE_LOGGING_HXX
+
+#include <windows.h>
+
+#include <TraceLoggingProvider.h>
+
+#include <plog/Appenders/IAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+TRACELOGGING_DECLARE_PROVIDER(g_hplogTraceLoggingProvider);
+
+namespace plog
+{
+
+template <class Formatter>
+class TraceLoggingAppender : public IAppender
+{
+public:
+    TraceLoggingAppender()
+    {
+        TraceLoggingRegister(g_hplogTraceLoggingProvider);
+    }
+
+    ~TraceLoggingAppender()
+    {
+        TraceLoggingUnregister(g_hplogTraceLoggingProvider);
+    }
```

Comment:
I don't think it's appropriate for it to be in a source file because it's part of a templated class, which by nature doesn't generate any source code until it is instantiated with a template parameter